app_dart/lib/src/model/appengine/task.dart

```diff
@@ -129,6 +131,79 @@ class Task extends Model<int> {
     );
   }
 
+  /// Creates a task based on a buildbucket build.
+  static Future<Task> fromBuildbucketResult(
+    bb.Build build,
+    DatastoreService datastore,
+  ) async {
+    log.fine("Creating task from buildbucket result: ${build.toString()}");
+
+    final String repository = build.input!.gitilesCommit!.project!.split('/')[1];
+    log.fine("Repository: $repository");
+
+    final String branch = build.input!.gitilesCommit!.ref!.split('/')[2];
```

Comment:
Same comment with this guy too. But this one is less intuitive. Also you might want to add in a comment the format of the thing you are trying to split. I am guessing it is a string separated by '/'.